test/check-machines-storage-pools

```diff
@@ -748,6 +748,242 @@ class TestMachinesStoragePools(VirtualMachinesCase):
+                    else:
+                        # Check when no volume
+                        b.wait_not_present("#storage-pool-delete-volumes")
+                        b.wait_in_text(".pf-c-modal-box__body div",
+                                       "No volumes exist in this storage pool.")
+                else:
+                    # Check the content of inactive pool deletion dialog
+                    b.wait_text(".pf-c-modal-box__body div",
+                                "Deleting an inactive storage pool will only undefine the pool. Its content will not be deleted.")
+
+                b.click('.pf-c-modal-box__footer button:contains("Delete")')
+
+                if self.xfail:
+                    b.wait_visible("div.pf-c-modal-box")
+                    if self.xfail_error:
+                        # Check the error shown on the footer of the deletion of dialog
+                        b.wait_in_text("div.pf-c-modal-box .pf-c-alert__description", self.xfail_error)
+                    # Close the dialog
+                    b.click('.pf-c-modal-box__footer button:contains("Cancel")')
+                    # Clean the pool if pool deletion is failed
+                    if self.activate:
+                        m.execute(f"virsh pool-destroy {self.name}")
+                    m.execute(f"virsh pool-undefine {self.name}")
+
+                b.wait_not_present("div.pf-c-modal-box")
+                self.test_obj.waitPoolRow(self.name, self.connection, False)
+
+            def verify(self):
+                if self.activate:
+                    if self.remove_volume and self.vol_name:
+                        if self.pool_type == "netfs":
+                            wait(lambda: self.vol_name not in m.execute(f'ls -A {self.source_path}'))
+                        elif self.pool_type == "dir":
+                            wait(lambda: self.vol_name not in m.execute(f'ls -A {self.target}'))
+                        elif self.pool_type == "disk":
+                            wait(lambda: self.vol_name not in m.execute("lsblk"))
+                    elif (not self.remove_volume) and self.vol_name:
+                        if self.pool_type == "netfs":
+                            vol_path = os.path.join(self.source_path, self.vol_name)
+                            wait(lambda: m.execute(f'test -f {vol_path}'))
+                        elif self.pool_type == "dir":
+                            vol_path = os.path.join(self.target, self.vol_name)
+                            wait(lambda: m.execute(f'test -f {vol_path}'))
+                        elif self.pool_type == "disk":
+                            wait(lambda: self.vol_name in m.execute("lsblk"))
+
+                # general check
+                self.test_obj.waitPoolRow(self.name, present=False)
+                wait(lambda: self.name not in m.execute("virsh pool-list --all"))
+
```

Comment:
How about also testing a deletion of dir storage pools? That way, we can remove a lot of code from testStoragePools, making this file a bit shorter :)